tools/toolchain/scripts/stage6/install_deepmd.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash -e
+
+# TODO: Review and if possible fix shellcheck errors.
+# shellcheck disable=SC1003,SC1035,SC1083,SC1090
+# shellcheck disable=SC2001,SC2002,SC2005,SC2016,SC2091,SC2034,SC2046,SC2086,SC2089,SC2090
+# shellcheck disable=SC2124,SC2129,SC2144,SC2153,SC2154,SC2155,SC2163,SC2164,SC2166
+# shellcheck disable=SC2235,SC2237
+
+[ "${BASH_SOURCE[0]}" ] && SCRIPT_NAME="${BASH_SOURCE[0]}" || SCRIPT_NAME=$0
+SCRIPT_DIR="$(cd "$(dirname "$SCRIPT_NAME")/.." && pwd -P)"
+
+deepmd_ver="2.0.0.b1"
+deepmd_sha256="4860d72ac8ecd485a501a51a8b3e85f1a62ecf477b164ad636b5824a63c83750"
+
+case "$DEEPMD_MODE" in
+  cpu)
+    tfcc_ver="2.3.0-cpu_cudaNone_0"
+    tfcc_sha256="b1b6e5c68a58e516178e6e855a70dd020021d1cf22a94ad291bb8e7541ec58db"
+    ;;
+  cuda)
+    tfcc_ver="2.3.0-gpu_cuda10.1_0"
+    tfcc_sha256="085ba903c76adbbcb57dcdb474e5da5fd36324ca64ce673acacf122d5a2f2b3d"
+    ;;
+esac
+
+cub_link="https://github.com/NVlabs/cub/archive/c3cceac115c072fb63df1836ff46d8c60d9eb304.zip"
```

Comment:
I believe CUB is now [included](https://docs.nvidia.com/cuda/cub/index.htm) in the regular CUDA Toolkit. So, there is no need to install it separately anymore.

Reply:
Well, in [DeePMD-kit repository](https://github.com/deepmodeling/deepmd-kit), they just use [the very fork](https://github.com/NVIDIA/cub/tree/b229817e3963fc942c7cc2c61715a6b2b2c49bed) of CUB as submodule, while removing it error would be raised when compiling. So here I tried to add it to DeePMD-kit code by downloading code archive file directly. According to your description, it might be OK building without it, and I will have a try.

Reply:
> I believe CUB is now [included](https://docs.nvidia.com/cuda/cub/index.htm) in the regular CUDA Toolkit. So, there is no need to install it separately anymore.

In their latest commits, they remove extra CUB requirement for CUDA>=11, shown [here](https://github.com/deepmodeling/deepmd-kit/commit/9b5e2b9b4d1c1ad53b1e6887177aab186fe412cf#diff-735c1a60650597d4822c37e3c7ea5157250a5baea1403120e26f73973ff6630c), as it has been included in CUDA Toolkit. However, for former CUDA versions, it might still be needed.

Reply:
Ok, then let's keep it for a year or so until everybody has upgrade to CUDA >= 11.